packages/klighd-core/src/views-rendering.tsx

```diff
@@ -964,40 +1111,41 @@ export function renderKRendering(kRendering: KRendering,
                     verticalAlignment: VerticalAlignment.CENTER
                 } as KVerticalAlignment
                 boundsAndTransformation.bounds = {
-                    x: calculateX(0, originalWidth, styles.kHorizontalAlignment, textWidth),
-                    y: originalHeight * 0.5,
-                    width: originalWidth,
-                    height: originalHeight
+                    x: calculateX(0, originalBounds.width, styles.kHorizontalAlignment, textWidth),
+                    y: originalBounds.height * 0.5,
+                    width: originalBounds.width,
+                    height: originalBounds.height
                 }
             } else {
                 // Offsets are already applied in the transformation, so set them to 0 here.
                 boundsAndTransformation.bounds = {
                     x: 0,
                     y: 0,
-                    width: originalWidth,
-                    height: originalHeight
+                    width: originalBounds.width,
+                    height: originalBounds.height
                 }
             }
-            if (providingRegion) {
-                // Store exact height of title text
-                providingRegion.regionTitleHeight = newHeight
-                providingRegion.regionTitleIndentation = newX
-            }
+
+            // Don't draw if the rendering is an empty KText
+            const isEmptyText = kRendering.type === K_TEXT && (kRendering as KText).text === ""
+
             // Draw white background for overlaying titles
-            if (context.depthMap && kRendering.isNodeTitle && ((providingRegion && providingRegion.detail === DetailLevel.FullDetails) || !providingRegion)
-                && kRendering.calculatedBounds && kRendering.calculatedBounds.height * context.viewport.zoom <= titleScalingFactorOption * kRendering.calculatedBounds.height
-                // Don't draw if the rendering is an empty KText
-                && (kRendering.type !== K_TEXT || (kRendering as KText).text !== "")) {
-                overlayRectangle = <rect x={0} y={0} width={originalWidth} height={originalHeight} fill="white" opacity="0.8" stroke="black"/>
+            if ((!providingRegion || providingRegion.detail === DetailLevel.FullDetails) && tooSmall && !isEmptyText) {
```

Comment:
thanks, these conditions do look a lot nicer with all individual parts in their own variables